src/moar.h

```diff
@@ -91,10 +92,13 @@ typedef double   MVMnum64;
 #  define MVM_USED_BY_JIT
 #endif
 
-/* stuff for uthash */
-#define uthash_fatal(msg) MVM_exception_throw_adhoc(tc, "internal hash error: " msg)
+/* Hashes */
+#define HASH_DEBUG_ITER 1
```

Comment:
Should this be on when we merge?

Reply:
Er, yes, bother, no. As in, no it should not be on when we merge.

Although I **did** wonder about whether the "throw an error when both iterating and modifying the same hash" should be enabled by default. As in go from what we have:

```
typedef struct {
    MVMuint32 pos;
#if HASH_DEBUG_ITER
    MVMuint32 serial;
    MVMuint64 owner;
#endif
}  MVMStrHashIterator;
```
to this instead

```
typedef struct {
    MVMuint32 pos;
    MVMuint32 serial;
#if HASH_DEBUG_ITER
    MVMuint64 owner;
#endif
}  MVMStrHashIterator;
```

and default to something like this error we have disabled by default in `uthash.h`:

```
#if MVM_HASH_THROW_ON_ITER_AFTER_ADD_KEY
    /* Warn if the user has been *caught* inserting keys during an iteration. */
    if (ht->bucket_rand_last != ht->bucket_rand)
        MVM_exception_throw_adhoc(tc, "Warning: trying to insert into the hash while iterating has undefined functionality\n");
#endif
```

except that we can also do it on naughty deletes (ie not at the iterator's current position)

Reply:
I guess that'd be a language-visible change, so we'd want to proceed with caution. Perhaps something to see if we can get away with after we know that shipping the new hash implementation hasn't had unexpected fallout.